### ISSUE TITLE
AbbreviationCoordinator#abbreviateUntilTextFits NPE

### ISSUE THREAD

**Issue by Guardiola31337** [mapbox/mapbox-navigation-android#1261]
Run into the following 💥 while testing https://github.com/mapbox/mapbox-navigation-android/pull/1233

```
09-05 15:35:54.437 26951-26951/com.mapbox.services.android.navigation.testapp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.mapbox.services.android.navigation.testapp, PID: 26951
    java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
        at com.mapbox.services.android.navigation.ui.v5.instruction.AbbreviationCoordinator.abbreviateUntilTextFits(AbbreviationCoordinator.java:74)
        at com.mapbox.services.android.navigation.ui.v5.instruction.AbbreviationCoordinator.abbreviateBannerText(AbbreviationCoordinator.java:65)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionLoader.getAbbreviatedBannerText(InstructionLoader.java:98)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionLoader.setText(InstructionLoader.java:93)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionLoader.loadInstruction(InstructionLoader.java:54)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.updateDataFromBannerInstruction(InstructionView.java:873)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.access$100(InstructionView.java:80)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$2.onChanged(InstructionView.java:199)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$2.onChanged(InstructionView.java:195)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at android.arch.lifecycle.MutableLiveData.setValue(MutableLiveData.java:33)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.updateBannerInstruction(NavigationViewModel.java:413)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.access$900(NavigationViewModel.java:53)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel$3.onMilestoneEvent(NavigationViewModel.java:306)
        at com.mapbox.services.android.navigation.v5.navigation.NavigationEventDispatcher.onMilestoneEvent(NavigationEventDispatcher.java:137)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorThreadListener.onMilestoneTrigger(RouteProcessorThreadListener.java:46)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorHandlerCallback$1.run(RouteProcessorHandlerCallback.java:98)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

It seems like a race condition with `abbreviations` from `AbbreviationCoordinator` getting out of sync somehow i.e. like mutating its state in parallel, while `abbreviateUntilTextFits` is being executed.

**Comment by Guardiola31337** [comment]
@abhishek1508 is this still an issue in the `1.0` UI SDK?

**Comment by abhishek1508** [comment]
Let's verify.
cc @JunDai @cafesilencio @langsmith

**Comment by cafesilencio** [comment]
I tested with the CustomePuckActivity duplicating the code snippet I found here -> https://github.com/mapbox/mapbox-navigation-android/pull/1233#issuecomment-421344608

I added the code to the willDisplay method in the activity.

I don't experience any crashes or errors with using the ".abbreviation()" that was indicated as the cause of the error.  As I"m unable to reproduce this in the 1.0 version of the SDK I'm closing this.
